### DISCUSSION TITLE
Simplify StreetComplete questions

### DISCUSSION THREAD

**Discussion by westnordost** [streetcomplete/StreetComplete#3862] [Ideas]
Specifically because of (#3857 / #3860) but basically because of the long ongoing trend to include a potential name / feature type name in the question for POIs, I have been thinking if this could not be streamlined. Maybe like this:

![simplify_questions](https://user-images.githubusercontent.com/4661658/157507235-85619ea4-75a0-4416-8309-6259e289d154.png)

**Disadvantages**:
- name and feature name of POI is less prominent, might be missed by users
- possibly less flexibility

**Advantages**:
- simplify code because it can be generic. Less code for current and future quests, thus easier to maintain.
- much less strings to translate and maintain for translators
- translators very often get those placeholders "%1$s" etc. wrong, finding and fixing such mistakes is quite time intensive
- more consistent user interface
- question can be more concise, i.e. less space wasted and less text for user to read

For me, the advantages somewhat outweigh the disadvantages. 

Opinions?

**Comment by Helium314** [comment]
I like it, especially because it's more consistent

> might be missed by users

Or maybe this (the change and the longer text) might teach users to look at the text above the question more often, leading to fewer notes about "not existing" things that are actually underground.

**Comment by riQQ** [comment]
I like it.

Small adjustment/improvement:
I would put the name first if there is one as it will be the most prominent "feature":
name: "Apostel-Apotheke (Pharmacy Counter) - house number 20"
no name: "house number 20"

**Reply by westnordost** [reply]
"house number" is at the same place as "on level 3" would be displayed. I think it is most important to first inform where it is located

**Comment by matkoniecz** [comment]
In general it seems a good idea to me, that is one of quite complex parts that produces a lot of work.

--------------------------

I would consider making this text more visible (larger font size/closer to black).

And split into lines, rather than having everything in one.

And maybe it would be better below the main question? Rather than above? It seems a better place for extra info.

--------------------------

And in general it would be possible to implement it, release and do not delete freed strings immediately - just in case that it turns to be a really bad idea after all.

**Reply by mnalis** [reply]
Uh, it seems the old strings have disappeared from POeditor.com (and only `*_title2` are present now :cry:) 

Is it possible to get them back somehow, or do translators now have to retranslate that 30+ strings from scratch (instead of just copying and slightly modifying old ones?)

**Reply by matkoniecz** [reply]
At minimum you can look at old commit state but it is not accessible to translators.

Polish: https://github.com/streetcomplete/StreetComplete/blob/9c8a96249bc4711a6306477b10b8f0e0d56b939a/app/src/main/res/values-pl/strings.xml

Croatian: https://github.com/streetcomplete/StreetComplete/blob/9c8a96249bc4711a6306477b10b8f0e0d56b939a/app/src/main/res/values-hr/strings.xml

(difference is in `values-hr/strings.xml` / `values-pl/strings.xml` at the end)

**Comment by mnalis** [comment]
I get all the benefits, but I'm still somewhat reserved about reaction of users, as the important parts would be less visible. 

I unfortunately do not share @Helium314 optimism about this feature inciting users to put more attention to "small print" and am afraid it is more likely to cause problems (in SC we have seen time and again that users - even experienced ones - just skip over such texts that the brain interprets like "repeating filler text of no importance"). To combat such unwanted outcome of user frustration, I'd:

- I'd put the **name** of the amenity (if present) **in bold** - as that is first _unique_ part that the user will look for, and it should thus stand out from "filler" unchanging text, as well from less-unique data like amenity type)
- I'd also just just the level number in bold too - it is easily overlooked as often not present, so should be highlighted.

e.g. it would look like "on level **3**: **Apostol-Apotheke** (Pharmacy Counter)"

I also agree with most other @matkoniecz suggestions:
- different formatting might show it is not standard "fillter text to ignore"
- split lines on logical boundaries and not randomly in the middle of the text (that makes it look even more like unimportant filler text; _and_ makes it harder to read)
- do not burn the bridge (do not erase the texts for few releases), in case we need to retreat due to new format being harder for users.

**Reply by westnordost** [reply]
What makes you think that the text about location and name would more likely be read if it is below the question rather than below the question?

**Reply by westnordost** [reply]
>Because it is different location then above the quest, which usually (so far) contained predominantly instructions about how to solve the quest (eg. building type, surface quality, number of parking spaces etc), Thus (for existing users of course) our brains are already wired to ignore that blob of text above quest.

Ehm, no? The texts you mention are **BELOW** the quest question.

**Reply by mnalis** [reply]
me looking in _disbelief_ :rofl: 

You are correct of course: those texts are **above** the answers list, but **below** the question itself, and my brain just decided to merge those two facts together, sorry. :heart: 

The idea was that those should go in _different_ spot  then where the current texts are (which seems to be the case and I misread); and that unique information like name should be in **bold**.

**Reply by smichel17** [reply]
My hypothesis is that users are trained to **ignore light gray text.** So, I think it does not need to be bold, just **black** (or nearly so).

**Comment by arrival-spring** [comment]
Showing the name for the current item on the map (in the same way as nearby highlighted elements) could help users to notice the name.

I'm not suggesting this instead of putting the name on the quest form though.

**Reply by mnalis** [reply]
I think SC already does it, but **only** if the quest being solved is matching `IS_SHOP_OR_DISUSED_SHOP_EXPRESSION` (or whatever else that quest highlights).  e.g.  if I'm solving `opening_hours` quest for `copyshop`, its name is shown, but if I'm solving the same quest for `tailor` or `office`, it is not.

It would be nice if `name`/`brand`/`operator` of **_currently selected element_** were always shown on the map too.

**Reply by arrival-spring** [reply]
See screenshot above, and I think it's this bit here:
https://github.com/streetcomplete/StreetComplete/blob/0e900879c915be0d2ab0ee4df234b3a338447711/app/src/main/java/de/westnordost/streetcomplete/map/MainFragment.kt#L887-L889

I don't think name of current object is ever shown

**Reply by mnalis** [reply]
ahhh, you seem to be correct, I tested on one area and it did show the same name... 

but that is due to https://www.openstreetmap.org/node/8105759231 and https://www.openstreetmap.org/node/1946566962 being at same position with same name :smile:  (it is actually the same shop, but as OSM does not _really_ allow combining `shop=stationery` and `shop=copyshop` into same node, it's been mapped as two separate nodes at basically same position)

So yes, that would be good thing to add to also show currently selected node (maybe in different color, if displaying it the same as other nodes led to it not being shown?)
